File: DataBindingSample/DataBindingSample/MainData.cpp
#include "pch.h"
#include "MainData.h"

using namespace Platform;
using namespace DataBindingSample;

MainData::MainData()
{
	m_MyTextValue = "MyText";
	m_MyFontSize = 12;
}

String^ MainData::MyTextValue::get()
{
	return m_MyTextValue;
}

void MainData::MyTextValue::set(String^ myTextValue)
{
	if (m_MyTextValue != myTextValue)
	{
		m_MyTextValue = myTextValue;
		NotifyPropertyChanged("MyTextValue");
	}
}

int MainData::MyFontSize::get()
{
	return m_MyFontSize;
}

void MainData::MyFontSize::set(int myFontSize)
{
	if (m_MyFontSize != myFontSize)
	{
		m_MyFontSize = myFontSize;
		NotifyPropertyChanged("MyFontSize");
	}
}

void MainData::NotifyPropertyChanged(String^ propertyName)
{
	PropertyChangedEventArgs^ args = ref new PropertyChangedEventArgs(propertyName);
	PropertyChanged(this, args);
}

File: DataBindingSample/DataBindingSampleCSharp/MainData.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DataBindingSampleCSharp
{
    public class MainData : INotifyPropertyChanged
    {
        #region private members
        private string _myTextValue = "MyText";
        private int _myFontSize = 12;
        #endregion

        #region public properties
        public string MyTextValue
        {
            get 
            { 
                return _myTextValue; 
            }
            set 
            {
                if (_myTextValue == value) return;
                _myTextValue = value;
                NotifyPropertyChanged("MyTextValue");
            }
        }

        public int MyFontSize
        {
            get
            {
                return _myFontSize;
            }
            set
            {
                if (_myFontSize == value) return;
                _myFontSize = value;
                NotifyPropertyChanged("MyFontSize");
            }
        }
        #endregion

        #region Implementation of INotifyPropertyChanged
        public event PropertyChangedEventHandler PropertyChanged;

        private void NotifyPropertyChanged(string propertyName)
        {
            if (PropertyChanged != null)
                PropertyChanged(this, new PropertyChangedEventArgs(propertyName));
        }
        #endregion
    }
}
